mautic/mautic
# Integration with Google Tag Manager

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" 
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/155
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:

### Integration Google Tag Manager on Mautic
In Setting Page -> Tracking Config i added a new panel with Google Tag Manager options.

Options includes:
* Google Tag Manager ID
* Enabled on your tracking page
* Enable on Mautic landing page

If ***Enabled on your tracking page*** is on, gtm is loaded on external page if user add mautic tracking code.

If ***Enable on Mautic landing page*** is on, gtm is loaded on all Mautic Landing Pages.

GTM script is loaded from backend, for external sites instead is loaded by mautic tracking code.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Settings -> Tracking Config
3. Edit Google Tag Manager fields, add a valid Google Tag Manager ID and enable the two checkboxes
4. Create a Landing Page
5. View the Landing Page and check if GTM is loaded




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9493"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
